17-switch-mass-hypo.md

```diff
@@ -10,5 +10,81 @@ minutes: 10
 > * Create a new selection starting from the stripping line output
 > * Change the particle hypothesis made by the stripping line
 
+There are many situations where you want to repurpose a stripping line to look for a new decay that is simliar in topology but dstinct from what was put into the stripping line. The easiest thing to do is, to change some of the hypothesis on particle IDs made in the stripping. This lesson will show you how to do that.
 
+> ## Reinterpreting stripping selections {.callout}
+> Note that with this method you can never find more candidates than the stripping has already found. 
+> However, you can reinterpret parts of the decay to look for new decay modes.
 
+As an example we will switch the decay of the D0 from (K pi) to (pi pi). 
+
+There is an algorithm that allows us to replace parts of the decay descriptor called SubstitutePID:
+
+```python
+# configure an algorithm to substitute the Kaon in the D0-decay by a second pion 
+from Configurables import SubstitutePID
+Subs = SubstitutePID(
+        'MakeD02pipi',
+        Code = "DECTREE('[D*(2010)+ -> (D0 -> K- pi+) pi+]CC')",
+        Substitutions = { # note that SubstitutePID can't handle automatic CC
+        'Charm -> (D0 -> ^K- pi+) Meson' : 'pi-', 
+        'Charm -> (D0 -> ^K+ pi-) Meson' : 'pi+', 
+        }
+)
+```
+
+The algorithm is configured with a name, an initial selection - here simply a decay descriptor and a dictionary specifying the substitutions to be performed. The particles for which the replacement should be done have to be marked with a ^. Note that SubstitutePID does not automatically handle complex conjugation via the CC operator. Therefore you have to specify all substitutions explicitely. 
```

Comment:
I think the `Code` part is not very clear. Maybe you should cite the LoKi functors lesson and we can add it, or if not you have to explain that DECTREE is a functor and so on.
Also, minor, `Subs` is an instance so if we follow `pep8` I think it should start with lowercase.